gtsfm/utils/metrics.py

```diff
@@ -0,0 +1,197 @@
+    """Computes statistics for angle between translations and direction measurements. 
+
+    Args: 
+        i2Ui1_dict: List of translation direction measurements. 
+        wTi_list: List of estimated camera poses. 
+
+    Returns: 
+        A statistics dict of the metrics errors in degrees. 
+    """
+    angles = []
+    for (i1, i2) in i2Ui1_dict:
+        i2Ui1 = i2Ui1_dict[(i1, i2)]
+        angles.append(
+            comp_utils.compute_translation_to_direction_angle(
+                i2Ui1, wTi_list[i2], wTi_list[i1]))
+    return get_errors_statistics(angles)
+
+
+def compute_averaging_metrics(
+    i2Ui1_dict: Dict[Tuple[int, int], Unit3],
+    wRi_list: List[Optional[Rot3]],
+    wti_list: List[Optional[Point3]],
+    gt_wTi_list: List[Optional[Pose3]],
+) -> Dict[str, StatsDict]:
+    """Computes statistics of multiple metrics for the averaging modules. 
+
+    Specifically, computes statistics of: 
+        - Rotation angle errors before BA, 
+        - Translation distances before BA, 
+        - Translation angle to direction measurements, 
+
+    Estimated poses and ground truth poses are first aligned before computing metrics.    
+
+    Args: 
+        i2Ui1_dict: Dict from (i1, i2) to unit translation measurement i2Ui1. 
+        wRi_list: List of estimated rotations.         
+        wti_list: List of estimated translations.
+        gt_wTi_list: List of ground truth poses. 
+
+    Returns: 
+        Dict from metric name to a StatsDict. 
+
+    Raises: 
+        ValueError if lengths of wRi_list, wti_list and gt_wTi_list are not all same. 
+    """
+    if len(wRi_list) != len(wti_list) or len(wRi_list) != len(gt_wTi_list):
+        raise ValueError(
+            "Lengths of wRi_list, wti_list and gt_wTi_list should be the same.")
+
+    wTi_list = []
+    for (wRi, wti) in zip(wRi_list, wti_list):
+        wTi_list.append(Pose3(wRi, wti))
+    wTi_aligned_list = comp_utils.align_poses(wTi_list, gt_wTi_list)
+
+    def get_rotations_translations_from_poses(poses):
+        rotations = []
+        translations = []
+        for pose in poses:
+            if pose is None:
+                rotations.append(None)
+                translations.append(None)
+                continue
+            rotations.append(pose.rotation())
+            translations.append(pose.translation())
+        return rotations, translations
+
+    wRi_aligned_list, wti_aligned_list = get_rotations_translations_from_poses(
+        wTi_aligned_list)
+    gt_wRi_list, gt_wti_list = get_rotations_translations_from_poses(
+        gt_wTi_list)
+
+    metrics = {}
+    metrics['rotation_averaging_angle'] = compute_rotation_angle_metrics(
+        wRi_aligned_list, gt_wRi_list)
+    metrics[
+        'translation_averaging_distance'] = compute_translation_distance_metrics(
+            wti_aligned_list, gt_wti_list)
+    metrics[
+        'translation_to_direction_angle'] = compute_translation_angle_metrics(
```

Comment:
i would add that this is in degrees also:

`translation_to_direction_angle_deg`